client/lib/domains/get-domain-type-text.js

```diff
@@ -6,19 +11,19 @@ import { type as domainTypes } from './constants';
 export function getDomainTypeText( domain = {} ) {
 	switch ( domain.type ) {
 		case domainTypes.MAPPED:
-			return 'Mapped Domain';
+			return translate( 'Mapped Domain' );
 
 		case domainTypes.REGISTERED:
-			return 'Registered Domain';
+			return translate( 'Registered Domain' );
 
 		case domainTypes.SITE_REDIRECT:
-			return 'Site Redirect';
+			return translate( 'Site Redirect' );
 
 		case domainTypes.WPCOM:
-			return 'Default Site Domain';
+			return translate( 'Default Site Domain' );
 
 		case domainTypes.TRANSFER:
-			return 'Transfer';
+			return translate( 'Transfer', { context: 'noun', comment: 'A transfer of a domain' } );
```

Comment:
:warning: This change will be queued for retranslation. We'll display the English text until then.